osivalidator/osi_general_validator.py

```diff
@@ -126,16 +130,19 @@ def main():
         DATA.cache_messages_in_index_range(first_of_blast, last_of_blast)
         MESSAGE_CACHE.update(DATA.message_cache)
 
-        # Launch computation
-        # TODO Change this to parallel processing
-        for i in range(first_of_blast, last_of_blast):
-            process_timestep(i)
-        # pool.map(process_timestep, range(first_of_blast, last_of_blast))
-
-        LOGGER.flush(LOGS)
-
+        if args.parallel:
+            # Launch parallel computation
+            # Recreate the pool
+            pool = Pool()
+            pool.map(process_timestep, range(first_of_blast, last_of_blast))                      
```

Comment:
How is an abort of the program handled here?

Reply:
You mean how it is closed or how to handle when an exception occurs? There is a `close_pool` function call for terminating after the line `pool.map` (see [here](https://github.com/OpenSimulationInterface/osi-validation/blob/rule-naming-update/osivalidator/osi_general_validator.py#L138)). For exception handling there needs to be implemented a try catch if something goes wrong during processing.

Reply:
Ok, fine thank you!